types/yargs/ts4.1/yargs-tests.ts

```diff
@@ -0,0 +1,1332 @@
+        .argv;
+}
+
+// From http://yargs.js.org/docs/#methods-coercekey-fn
+function Argv$coerce() {
+    const ya = yargs
+        .coerce('file', (arg: string) => {
+            return fs.readFileSync(arg, 'utf8');
+        })
+        .argv;
+}
+function Argv$coerces() {
+    const ya = yargs
+        .coerce({
+            date: Date.parse,
+            json: JSON.parse
+        })
+        .argv;
+}
+function Argv$coerceWithKeys() {
+    const ya = yargs
+        .coerce(['src', 'dest'], path.resolve)
+        .argv;
+}
+
+// From http://yargs.js.org/docs/#methods-failfn
+function Argv$fail() {
+    const ya = yargs
+        .fail((msg, err, { help }) => {
+            if (err) throw err; // preserve stack
+            console.error('You broke it!');
+            console.error(msg);
+            console.error(help());
+            process.exit(1);
+        })
+        .argv;
+
+    // Providing false as a value for fn can be used to prevent yargs from
+    // exiting and printing a failure message
+    const ya2 = yargs
+        .fail(false)
+        .argv;
+}
+
+function Argv$implies() {
+    const ya = yargs
+        .implies('foo', 'snuh')
+        .implies({
+            x: 'y'
+        })
+        .argv;
+}
+
+function Argv$count() {
+    const ya = yargs
+        .count('size')
+        .count(['w', 'h'])
+        .argv;
+}
+
+function Argv$number() {
+    const ya = yargs
+        .number('n')
+        .number(['width', 'height'])
+        .argv;
+}
+
+function Argv$updateStrings() {
+    const ya = yargs
+        .command('run', 'the run command')
+        .help('help')
+        .updateStrings({
+            'Commands:': 'My Commands -->\n'
+        })
+        .wrap(null)
+        .argv;
+}
+
+function Argv$default() {
+    const ya = yargs
+        .default('random', function randomValue() {
+            return Math.random() * 256;
+        })
+        .argv;
+}
+
+function Argv$configObject() {
+    const ya = yargs
+        .config({ foo: 1, bar: 2 })
+        .argv;
+}
+
+function Argv$configParseFunction() {
+    const ya = yargs
+        .config('settings', (configPath) => {
+            return JSON.parse(fs.readFileSync(configPath, 'utf-8'));
+        })
+        .config('settings', 'description', (configPath) => {
+            return JSON.parse(fs.readFileSync(configPath, 'utf-8'));
+        })
+        .argv;
+}
+
+function Argv$helpDescriptionExplicit() {
+    const ya = yargs
+        .help('help', 'description', true)
+        .argv;
+}
+
+function Argv$showHelpConsoleLevel() {
+    yargs.showHelp("log"); // prints to stdout using console.log()
+}
+
+function Argv$getCompletion() {
+    const ya = yargs
+        .option('foobar', {})
+        .option('foobaz', {})
+        .completion()
+        .getCompletion(['./test.js', '--foo'], (completions) => {
+            console.log(completions);
+        })
+        .argv;
+}
+
+function Argv$getHelp() {
+	const ya = yargs.getHelp().then((help: string) => {
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/yargs/ts4.1/yargs-tests.ts;
  Line: 752;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.